lib/Echidna/Exec.hs

```diff
@@ -120,12 +123,28 @@ fuzz l n ts v p = do
   return $ listToMaybe [cs | (cs, passed) <- results, not passed]
     where run cs = p $ execState (forM_ cs execCall) v
 
+checkTest :: PropertyType -> VM -> Text -> Bool
+checkTest ShouldReturnTrue             = checkBoolExpTest True
+checkTest ShouldReturnFalse            = checkBoolExpTest False
+checkTest ShouldRevert                 = checkRevertTest
+checkTest ShouldReturnFalseRevert      = checkFalseOrRevertTest
 
-checkETest :: VM -> Text -> Bool
-checkETest v t = case evalState (execCall (t, [])) v of
-  VMSuccess (B s) -> s == encodeAbiValue (AbiBool True)
+checkBoolExpTest :: Bool -> VM -> Text -> Bool
+checkBoolExpTest b v t = case evalState (execCall (t, [])) v of
+  VMSuccess (B s) -> s == encodeAbiValue (AbiBool b)
   _               -> False
 
+checkRevertTest :: VM -> Text -> Bool
+checkRevertTest v t = case evalState (execCall (t, [])) v of
+  (VMFailure Revert) -> True
+  _                  -> False
+
+checkFalseOrRevertTest :: VM -> Text -> Bool
```

Comment:
Why is there a `checkFalseOrRevert` but no `checkTrueOrRevert`?

Reply:
checkFalseOrRevert is useful to implement [the zoo property standard](https://github.com/trailofbits/smart-contracts-testing/tree/master/zoo/properties#return-value).

Reply:
Let's definitely keep it then, but maybe add a checkTrueOrRevert for completeness?